frontend/src/app/application-forms/validators/number-validation.ts

```diff
@@ -1,14 +1,25 @@
 import { AbstractControl, ValidatorFn } from '@angular/forms';
 
-export function numberValidator(): ValidatorFn {
+export function numberValidator(allowDash: boolean = false): ValidatorFn {
   return (control: AbstractControl): { [key: string]: any } => {
-    const val = control.value;
-    if (val && val.length) {
+    if (allowDash) {
+      const val = control.value;
+      if (val && val.length) {
+        const numberRegex = /^[\d -]+$/;
+        const valid = numberRegex.test(val);
+        return valid ? null : { numberDashRequirement: true };
+      } else {
+      return null;
```

Comment:
incorrect indentation here

Reply:
Indentation should look something like this 

```
export function numberValidator(allowDash: boolean = false): ValidatorFn {
  return (control: AbstractControl): { [key: string]: any } => {
    if (allowDash) {
      const val = control.value;
      if (val && val.length) {
        const numberRegex = /^[\d -]+$/;
        const valid = numberRegex.test(val);
        return valid ? null : { numberDashRequirement: true };
      } else {
        return null;
      }
    } else {
      const val = control.value;
      if (val && val.length) {
        const numberRegex = /^\d+$/;
        const valid = numberRegex.test(val);
        return valid ? null : { numberRequirement: true };
      } else {
        return null;
      }
    }
  };
}
```